flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/ZooKeeperCompletedCheckpointStoreMockitoTest.java

```diff
@@ -89,46 +77,72 @@ public void testCheckpointRecovery() throws Exception {
 		expectedCheckpointIds.add(1L);
 		expectedCheckpointIds.add(2L);
 
-		final RetrievableStateHandle<CompletedCheckpoint> failingRetrievableStateHandle = mock(RetrievableStateHandle.class);
-		when(failingRetrievableStateHandle.retrieveState()).thenThrow(new IOException("Test exception"));
-
-		final RetrievableStateHandle<CompletedCheckpoint> retrievableStateHandle1 = mock(RetrievableStateHandle.class);
-		when(retrievableStateHandle1.retrieveState()).then(
```

Comment:
are all these refactorings necessary for the test to run on jdk 9? If not, please move them into a separate commit.